### ISSUE TITLE
Minio S3 Redirect (private ACL) uploads, but does not use correct domain

### ISSUE THREAD

**Issue by tankerkiller125** [outline/outline#5352]
**To Reproduce**
1. Set up Outline to use Minio file storage (path style)
2. Set the ACL to be private
3. Attempt to upload a file (this part will work properly)
4. Attempt to view/download file (you will be taken to AWS instead of your own hosted minio instance)

**Expected behavior**
Redirect should use the Minio hostname, not AWS S3

**Outline (please complete the following information):**
- Install: self-hosted
- Version: v0.69.2  (digest: sha256:0018ebc29673344c388ae51b16ca0899578e2e959a661cc670d760d8ed335d02)

**Desktop (please complete the following information):**
 - Browser: All
 - Version: All

**Mobile (please complete the following information):**
 - Device: All
 - OS: All
 - Browser: All

**Comment by tankerkiller125** [comment]
I ended up solving this by using the same Minio URL as an AWS accelerate URL, and then using a cloudflare worker to rewrite the double path to the correct pathing. But the issue still remains in the code.

**Comment by cuong-tran** [comment]
I was able to setup self-hosted Outline version 0.69.2 with self-hosted MinIO and it's working fine.
Mine **docker.env**:
`AWS_REGION=local
AWS_S3_ACCELERATE_URL=
AWS_S3_UPLOAD_BUCKET_URL=https://s3.<domain>
AWS_S3_UPLOAD_BUCKET_NAME=outline-bucket
AWS_S3_UPLOAD_MAX_SIZE=26214400
AWS_S3_FORCE_PATH_STYLE=true
AWS_S3_ACL=private`

In your case, I'm wondering where does the Outline get AWS URL from?

**Comment by Coksnuss** [comment]
I experience the same issue when using my S3 storage on the scaleway cloud. Settings are like this:

```
- AWS_ACCESS_KEY_ID=SCW<KEY>
- AWS_SECRET_ACCESS_KEY=<secret>
- AWS_REGION=fr-par
- AWS_S3_UPLOAD_BUCKET_URL=https://<mybucket>.s3.fr-par.scw.cloud
- AWS_S3_UPLOAD_BUCKET_NAME=<mybucket>
- AWS_S3_UPLOAD_MAX_SIZE=26214400
- AWS_S3_FORCE_PATH_STYLE=false
- AWS_S3_ACL=private
```

Uploading works just fine. But when trying to download an attachment the request goes like this:

`GET https://<mydomain>.<tld>/api/attachments.redirect?id=80518e34-b2ef-41c5-bc8c-df2611e184bf`
Status: 302 Found
Location: https://mybucket.s3.fr-par.amazonaws.com/uploads/5f2d6bd8-d813-41ea-984c-e89abcad8e22/80518e34-b2ef-41c5-bc8c-df2611e184bf/myfile?X-Amz-Algorithm=<removed-for-confidentiality>&X-Amz-SignedHeaders=host&response-content-disposition=attachment

So unfortunately, the wrong domain is used (amazonaws.com as opposed to scw.cloud). Otherwise everything looks correct.

**Comment by tankerkiller125** [comment]
I'm using the path style already and I'm getting this issue.

```
AWS_REGION=local
AWS_ACCESS_KEY_ID=<key>
AWS_SECRET_ACCESS_KEY=<key>
AWS_S3_UPLOAD_BUCKET_URL=https://data.scobo.dev/<bucket>
AWS_S3_UPLOAD_BUCKET_NAME=<bucket>
AWS_S3_UPLOAD_MAX_SIZE=26214400
AWS_S3_FORCE_PATH_STYLE=true
AWS_S3_ACL=private
```

**Comment by tommoor** [comment]
Try adding `AWS_S3_ACCELERATE_URL` with the same value as `AWS_S3_UPLOAD_BUCKET_URL` as a workaround – this is all in dire need of rethinking, but there are thousands of existing installs out there with these configs now 😢

**Comment by tankerkiller125** [comment]
@tommoor This is what I'm already doing, as previously noted to make that work I had to use a cloudflare worker to write the duplicate path (e.g it uses /data/data instead of just /data) because using just (data.domain.tld) for the accelerate URL didn't work correctly.

**Comment by qhj** [comment]
I can confirm the duplicate path issue.

**Comment by Nicicalu** [comment]
It could maybe help to look at my answer in this issue: https://github.com/outline/outline/discussions/3366#discussioncomment-3077478

**Comment by tommoor** [comment]
closed by 74722b80f2b71703340297ceb2c36104bcd2ca87
